pandas/tests/tools/test_to_datetime.py

```diff
@@ -609,6 +609,121 @@ def test_to_datetime_dtarr(self, tz):
         result = to_datetime(arr)
         assert result is arr
 
+    @pytest.mark.parametrize(
+        "data, expected",
+        [
+            (
+                Timestamp("2001-10-01 12:00:01.123456789"),
+                Timestamp("2001-10-01 12:00:01.123456"),
+            ),
+            (
+                datetime(2002, 10, 1, 12, 00, 1, 123456),
+                Timestamp("2002-10-01 12:00:01.123456"),
+            ),
+            ("10/01/03 12:00:01.123456789", Timestamp("2003-10-01 12:00:01.123456789")),
+        ],
+    )
+    def test_to_datetime_preserves_resolution_when_possible(self, data, expected):
+        # GH 49298
+        if not isinstance(data, str):
+            result = to_datetime([data])
+            tm.assert_equal(result, DatetimeIndex([data]))
+
+        result = to_datetime([data], format="%m/%d/%y %H:%M:%S.%f")
+        tm.assert_equal(result, DatetimeIndex([expected]))
+
+    @pytest.mark.parametrize(
+        "init_constructor, end_constructor",
+        [
+            (Index, DatetimeIndex),
+            (list, DatetimeIndex),
+            (np.array, DatetimeIndex),
+            (Series, Series),
+        ],
+    )
+    def test_to_datetime_arraylike_contains_pydatetime_and_timestamp(
```

Comment:
not sure I follow, why is this (and the following) test necessary? I'd have thought that just some example which checks `to_datetime(arg, format=` where the input list contains both strings and `datetime.datetime` objects and `Timestamp` objects and `format` is specified.
Not sure I see why the constructor tests add, but I might be missing something

I'd also suggest adding a test where `format` is ISO8601 (as that goes down a different path), e.g. `%Y-%m-%d`

Reply:
This test performs the comparison you mention: input with different types while verifying that the output matches the expected format. Constructors should not be necessary (at least in this test) so I'll remove them. Also, I'll add a case where `format` is SO8601 here.

The `test_to_datetime_arraylike_contains_pydatetime_and_timestamp_utc` test aimed to check that the timezone was preserved as expected. It can be removed, but I wanted to check consistency across different input types